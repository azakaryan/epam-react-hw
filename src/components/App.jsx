import { Component } from 'react';
import React from 'react';
import { Route, Switch } from 'react-router';
import routes from '../routes';

export default class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Switch>
        {routes.map((route, i) => (
          <Route key={i} {...route} />
        ))}
      </Switch>
    );
  }
}
