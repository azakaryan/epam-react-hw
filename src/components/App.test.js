import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import App from './App';

expectEnzymify.extend(enzymify());


describe('App Component', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
  //
  // it('renders a App element', () => {
  //   const wrapper = shallow(<App />);
  //   expectEnzymify(wrapper).toBeA('Switch');
  // });
});
