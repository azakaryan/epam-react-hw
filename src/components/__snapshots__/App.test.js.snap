// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    key="0"
    path="/"
  />
  <Route
    component={[Function]}
    key="1"
    path="/movie/:id"
  />
</Switch>
`;
